src/cats.py

```diff
@@ -44,7 +46,7 @@
 
 FROZEN = False
 
-ROLES = {}
+ROLES: Dict[str, str] = {}
```

Comment:
This is wrong. You also put `Dict[str, str]` in a handful of other places where it's wrong. Actually check how the type is being used before throwing an arbitrary annotation on it. If you can't figure it out, leave it un-annotated.